### DISCUSSION TITLE
Catch all segments dynamic api routes

### DISCUSSION THREAD

**Discussion by rinvii** [Xunnamius/next-test-api-route-handler#989] [Q&A] [answered]
I'm not seeing any documentation related to [catch all api routes](https://nextjs.org/docs/pages/building-your-application/routing/api-routes#catch-all-api-routes). Is there support for this? Thank you so much for this amazing library :)

**Comment by Xunnamius** [comment, marked as answer]
Hi there! NTARH has support for all API route types. Testing catch-all routes shouldn't be too different from any other route, you just have to provide the expected [`params`](https://github.com/Xunnamius/next-test-api-route-handler?tab=readme-ov-file#paramspatcher-params) object.

Using the Next.js docs as an example (given URL `/api/post/a/b/c`):

```typescript
/* file: pages/api/post/[...slug].ts */
import type { NextApiRequest, NextApiResponse } from 'next'
 
export default function handler(req: NextApiRequest, res: NextApiResponse) {
  const { slug } = req.query
  res.end(`Post: ${slug.join(', ')}`)
}
```

```typescript
/* file: test/main.test.ts */
import { testApiHandler } from 'next-test-api-route-handler';
// Import the handler under test from the pages/api directory
import * as pagesHandler from '../pages/api/[...slug]';

it('does what I want', async () => {
  expect.hasAssertions();

  await testApiHandler({
    pagesHandler,
    params: { slug: ['a', 'b', 'c'] },
    test: async ({ fetch }) => {
      const res = await fetch({ method: 'POST' });
      await expect(res.text()).resolves.toBe('Post: a, b, c');
    }
  });
});
```
